File: problems/12-greedy/002-assign-cookies/java/Solution.java
import java.util.Arrays;

public class Solution {
    public int findContentChildren(int[] greed, int[] cookies) {
        Arrays.sort(greed);
        Arrays.sort(cookies);

        int child = 0;
        int cookie = 0;

        while (child < greed.length && cookie < cookies.length) {
            if (cookies[cookie] >= greed[child]) {
                child++;
            }
            cookie++;
        }

        return child;
    }
}

File: problems/12-greedy/002-assign-cookies/python/solution.py
class Solution:
    def findContentChildren(self, greed: list[int], cookies: list[int]) -> int:
        greed.sort()
        cookies.sort()

        child = 0
        cookie = 0

        while child < len(greed) and cookie < len(cookies):
            if cookies[cookie] >= greed[child]:
                child += 1
            cookie += 1

        return child
